### PR TITLE
Optimize circular buffer to avoid modulo[CLA: trivial]

### PR DESCRIPTION
**Pull request by AtariDreams**
[CLA: trivial]

Avoid doing the division via modulo where possible.

### CODE REVIEW

**Comment by t8m** [comment]
Unfortunately this is not acceptable with CLA: trivial. Could you please remove the CLA: trivial annotation and submit a regular CLA?

**Comment by AtariDreams** [comment]
> Unfortunately this is not acceptable with CLA: trivial. Could you please remove the CLA: trivial annotation and submit a regular CLA?

How about I only submit the first file then, since it is trivial?

File: crypto/bio/bio_dump.c [hunk 1]
```diff
@@ -141,9 +141,10 @@ int BIO_hex_string(BIO *out, int indent, int width, const void *data,
 
         BIO_printf(out, "%02X:", d[i]);
 
-        j = (j + 1) % width;
-        if (!j)
+        if (++j >= width) {
+            j = 0;
```
Comment (tom-cosgrove-arm) [inline crypto/bio/bio_dump.c:145]: This is a needless optimisation - the execution speed here will be dominated by the calls to `printf()`.

It's worthwhile making this change if it makes the intent clearer, which I'm on the fence about - what do other people think?

Comment (mattcaswell) [reply crypto/bio/bio_dump.c:145]: I'm not sure it does particularly. It doesn't make it any worse, but I'm not sure it makes it better either.

Comment (t8m) [reply crypto/bio/bio_dump.c:145]: It at least corrects a coding style error as `if (!j)` should be written as `if (j == 0)`

**Review by t8m — APPROVED** [review]
OK with CLA: trivial

File: crypto/bio/bio_dump.c (hunk 1 above)
Comment (mattcaswell) [reply crypto/bio/bio_dump.c:145]: Yeah. Ok.

**Review by mattcaswell — APPROVED** [review]
Agree trivial

File: crypto/bio/bio_dump.c (hunk 1 above)
Comment (tom-cosgrove-arm) [reply crypto/bio/bio_dump.c:145]: That same style error is at line 139

**Review by mspncp — APPROVED** [review]
Agree with trivial.

Note: the commit message needs to be amended: the `[CLA: trivial]` does not belong in the subject line and the `CLA: trivial` line should be added at the end of the body.

```
Optimize circular buffer to avoid modulo

Avoid doing the division via modulo where possible.

CLA: trivial
```

This can be fixed by the OpenSSL team member who merges this pull request.

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**